Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs

```diff
@@ -481,6 +482,40 @@ public async Task NonRecoverableHttpError()
             await Assert.ThrowsExceptionAsync<TaskCanceledException>(() => call.ToTask());
         }
 
+        /// <summary>
+        /// Verify that channel will not crash with StatefulMutexException when calling setEnabled(false) method during sending logs.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(1)]
+        [DataRow(4)]
+        public async Task DisableChannelDuringSendingLogs(int countLogs)
+        {
+            try
+            {
+                // Prepare data.
+                MockStorage storage = new MockStorage();
+                var appSecret = Guid.NewGuid().ToString();
+                Channel channel = new Channel(ChannelName, MaxLogsPerBatch, new TimeSpan(), MaxParallelBatches,
+                    appSecret, _mockIngestion.Object, storage);
+                SetupEventCallbacks();
+
+                // Run methods in parallel
+                for (int i = 0; i < countLogs; i++)
+                {
+                    channel.EnqueueAsync(new TestLog()).RunNotAsync();
+                }
+                channel.SetEnabled(false);
+
+                // Wait when tasks will be finalized.
+                await Task.Delay(1500);
```

Comment:
Is there a way to work this delay around? For example `Task.WhenAny` with additional task wrapper for lines 496-507

Reply:
Yep, let's avoid delays

Reply:
We cannot avoid using delay because we use `RunNotAsync` and we need sure that method will be finished before finalizing the test.